File: python/src/azex/types/deposit.py
"""Types for the deposit and checkout resources."""

from __future__ import annotations

from typing import List, Literal, Optional

from pydantic import BaseModel


class ChainInfo(BaseModel):
    chain_id: int
    name: str
    confirmations_required: int


class DepositRecord(BaseModel):
    uid: str
    chain_id: int
    token: str
    amount: str
    tx_hash: str
    status: Literal["pending", "confirmed", "credited"]
    created_at: str
    confirmed_at: Optional[str] = None


class DepositInfo(BaseModel):
    deposit_address: str
    chains: List[ChainInfo]
    tokens: List[str]
    deposits: List[DepositRecord]


class CheckoutSession(BaseModel):
    uid: str
    chain_id: int
    currency_id: int
    amount: Optional[str] = None
    status: Literal["pending", "completed", "cancelled", "expired"]
    payment_address: Optional[str] = None
    created_at: str
    expires_at: Optional[str] = None

File: typescript/src/types/deposit.ts
export interface ChainInfo {
  chain_id: number;
  name: string;
  confirmations_required: number;
}

export interface DepositRecord {
  uid: string;
  chain_id: number;
  token: string;
  amount: string;
  tx_hash: string;
  status: "pending" | "confirmed" | "credited";
  created_at: string;
  confirmed_at?: string | null;
}

export interface DepositInfo {
  deposit_address: string;
  chains: ChainInfo[];
  tokens: string[];
  deposits: DepositRecord[];
}

export interface CheckoutSession {
  uid: string;
  chain_id: number;
  currency_id: number;
  amount?: string;
  status: "pending" | "completed" | "cancelled" | "expired";
  payment_address?: string;
  created_at: string;
  expires_at?: string;
}

export interface CheckoutCreateParams {
  chain_id: number;
  currency_id: number;
  amount?: string;
}
